media-manager-service/src/main/scala/routes/ServerRoutes.scala

```diff
@@ -194,16 +159,18 @@ private[service] final class ServiceRoutes(system: ActorSystem[_]) extends Spray
 		}
 	}
 
-	//need revise for play
 	val playFile: Route = path("play" / JavaUUID) { id =>
 		get {
-			complete("playFile")
-			// onComplete(fileActorHandler.play(id)) {
-			// 	case Success(Some(file)) => 
-			// 		complete(HttpResponse(entity = FileHandler.getChunked(s"${file.fileName}.${file.ext}")))
-			// 	case Success(None) => complete(s"Unable to find your file id: $id")
-			// 	case Failure(e) => complete(e.toString)
-			// }
+			onComplete(fileActorHandler.playFile(id)) {
+				case Success(value) => value match {
+					case FileActor.Play(sourceRef, contentTypeString, _) =>
```

Comment:
If `FileActor.Play` is a case class or pattern-match-able (i.e. it has an unapply method), you can just use nested destructuring:

```
case Success(FileActor.Play(sourceRef, contentTypeString, _)) =>
```

Reply:
resolved at [this commit](https://github.com/Try-Parser/MediaFileEncoder/pull/20/commits/7ee43c7d7d536ad1d132d52ab9fa2a8e29232ca5)